app/extensions/ipfs/ipfs.html

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta name="availableLanguages" content="">
+    <meta name="defaultLanguage" content="en-US">
+    <meta http-equiv="Content-Security-Policy" content="
+      default-src 'none';
+      form-action http://localhost:*;
+      script-src 'self' 'unsafe-eval';
+      img-src * data: file: chrome-extension:;
+      style-src 'self' 'unsafe-inline'; font-src 'self'; 
+      connect-src 'self'
+        wss://star-signal.cloud.ipfs.team
+        wss://ams-1.bootstrap.libp2p.io
+        wss://sfo-1.bootstrap.libp2p.io
+        wss://lon-1.bootstrap.libp2p.io
+        wss://sfo-2.bootstrap.libp2p.io
+        wss://sfo-3.bootstrap.libp2p.io
+        wss://sgp-1.bootstrap.libp2p.io
+        wss://nyc-1.bootstrap.libp2p.io
+        wss://nyc-2.bootstrap.libp2p.io
+      ;
+      object-src 'self'; plugin-types application/browser-plugin"/>
```

Comment:
In order to support a background page (and not just scripts) had to add manually all the endpoints a js-ipfs node connects on startup. This is a bit of annoyance as it will stop js-ipfs to connect dynamically to other public WebSockets endpoints unless the CSP can be relaxed.

Reply:
`wss://*`?

Reply:
oh wow, it actually takes wildcards. I completely had discarded that option.